checker/examples/units-extension/Demo.java

```diff
@@ -29,4 +37,26 @@ void auto(@s int time) {
         frq = 99 / time;
     }
 
+    public static void main(String[] args) {
+        @Hz int hertz = toHz(20);
+        @s int seconds = 5 * UnitsTools.s;
+
+        @SuppressWarnings("units")
+        @s(Prefix.milli) int millisec = 10;
+
+        @SuppressWarnings("hz")
+        @kHz int kilohertz = 30;
+
+        @Hz int resultHz = hertz + 20 / seconds;
+        System.out.println(resultHz);
+
+        @kHz int resultkHz = kilohertz + 50 / millisec;
+        System.out.println(resultkHz);
+
+        // this demonstrates the type hierarchy resolution: the common supertype of Hz and kHz is Frequency, so this statement will pass
+        @Frequency int okTernaryAssign = seconds > 10 ? hertz : kilohertz;
+
+        // on the other hand, this statement expects the right hand side to be a Hz, so it will fail
```

Comment:
The output is checked against Expected.txt as part of the make script. I can add //::error for documentation but this test isn't executed through the normal test framework due to the need to load the external units.